geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -190,20 +191,48 @@ object Transformers extends EnhancedTokenParsers with LazyLogging {
     override def dependenciesOf(fieldNameMap: Map[String, Field]): Seq[String] = e.dependenciesOf(fieldNameMap)
   }
   case class Cast2Int(e: Expr) extends CastExpr {
-    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Any =
-      e.eval(args).asInstanceOf[String].toInt
+    override def eval(args: Array[Any])(implicit ctx: EvaluationContext): Int =
+      e.eval(args) match {
+        case null => null.asInstanceOf[Int]
+        case i: Int => i
+        case d: Double => d.toInt
+        case f: Float => f.toInt
+        case l: Long =>  l.toInt
+        case a: Any => a.asInstanceOf[String].toInt
```

Comment:
change these to `a.toString.toInt` - it's a no-op on a string, but will handle other cases maybe